packages/flutter/lib/src/widgets/animated_switcher.dart

```diff
@@ -394,6 +395,6 @@ class _AnimatedSwitcherState extends State<AnimatedSwitcher> with TickerProvider
   @override
   Widget build(BuildContext context) {
     _rebuildOutgoingWidgetsIfNeeded();
-    return widget.layoutBuilder(_currentEntry?.transition, _outgoingWidgets!);
+    return widget.layoutBuilder(_currentEntry?.transition, _outgoingWidgets!.where((Widget outgoing) => outgoing.key != _currentEntry?.transition.key).toSet().toList());
```

Comment:
What does this look like?  Does the existing one just disappear to be replaced by the new one?

Reply:
Yes. If we were to display both versions simultaneously then we would have the issue with duplicate keys. The duplicate keys issue seems to not only be a problem with the LayoutIds but also locally for the AnimatedSwitcher.